source/MaterialXCore/Interface.cpp

```diff
@@ -92,10 +95,19 @@ OutputPtr PortElement::getConnectedOutput() const
     const string& outputString = getOutputString();
     OutputPtr result = nullptr;
 
+    // Want to find a node or nodegraph at the same level as the 
+    // parent element to this port. 
+    ConstElementPtr parent = getParent();
+    ConstElementPtr root = parent ? parent->getParent() : nullptr;
+
     // Look for an output in a nodegraph
     if (hasNodeGraphString())
     {
-        NodeGraphPtr nodeGraph = resolveRootNameReference<NodeGraph>(getNodeGraphString());
+        NodeGraphPtr nodeGraph = resolveRootNameReference<NodeGraph>(getNodeGraphString(), root);
```

Comment:
Check at sibling node level for node/nodegraph instead of just checking at the root which isn't actually correct as you could inadvertently find a node/nodegraph 2 levels removed. Though nested nodegraphs are not supported yet, this code should still behave correctly when it is.